src/IECoreMaya/SceneShapeInterface.cpp

```diff
@@ -1470,3 +1494,79 @@ SceneInterface::Path SceneShapeInterface::fullPathName( std::string relativeName
 	return fullPath;
 }
 
+bool SceneShapeInterface::readConvertParam( CompoundParameterPtr parameters, int attrIndex ) const
+{
+	// Read convert parameter from "queryConvertParameters" attribute and set it to parameters inplace.
+
+	IECorePython::ScopedGILLock gilLock;
+
+	// Create a Python list that has the tokenized string of the attribute value for parsing.
+	boost::python::list parserArgList;
+	{
+		MPlug pConvertParamQueries( thisMObject(), aConvertParamQueries );
+		MPlug pConvertParamQuery = pConvertParamQueries.elementByLogicalIndex( attrIndex );
+		MString paramsStr;
+		pConvertParamQuery.getValue( paramsStr );
+
+		const std::string pstr = paramsStr.asChar();
+		boost::tokenizer<boost::char_separator<char> > t( pstr, boost::char_separator<char>( " " ) );
+		boost::tokenizer<boost::char_separator<char> >::const_iterator it, endIt;
+		for ( it = t.begin(), endIt = t.end(); it != endIt; ++it )
+		{
+			parserArgList.append( *it );
+		}
+	}
+
+	boost::python::dict pyExecLocalsDict;
+	pyExecLocalsDict[ "parserArgList" ] = parserArgList;
+	pyExecLocalsDict[ "parameters" ] = parameters;
+
+	boost::python::object sysModules = boost::python::object( boost::python::handle<>( boost::python::borrowed( PyImport_GetModuleDict() ) ) );
```

Comment:
This looks more verbose than necessary to me. I think something along the lines of this should work :

```
import( "IECore" ).attr( "ParameterParser" )().attr( "parse" )( args, parameters );
```

Perhaps there is a good reason for the separate compile/execute approach that I don't know about, but it seems to me that speed would probably be a poor reason, given that going out to Python means we'd better not be needing performance anyway.